ZeeguuAPI/ZeeguuAPI/ZeeguuAPI.swift

```diff
@@ -0,0 +1,334 @@
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil) {
+				completion(langCode: response!)
+			} else {
+				completion(langCode: nil)
+			}
+		}
+	}
+	
+	/// Sets the language code of the native langugage of the logged in user.
+	///
+	/// - parameter newLanguageCode: The language code of the language that will be the new native language.
+	/// - parameter completion: A block that will receive a boolean indication if the request succeeded.
+	public func setNativeLanguage(newLanguageCode: String, completion: (success: Bool) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(success: false)
+		}
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.NativeLanguage, pathComponents: [newLanguageCode], method: HTTPMethod.POST, parameters: nil)
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil && response! == "OK") {
+				completion(success: true)
+			} else {
+				completion(success: false)
+			}
+		}
+	}
+	
+	/// Returns the language code of the learned and native langugage of the logged in user.
+	///
+	/// - parameter completion: A block that will receive a `JSON` object, which contains the dictornary with language codes of the learned and native language.
+	public func learnedAndNativeLanguage(completion: (dict: JSON?) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(dict: nil)
+		}
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.LearnedAndNativeLanguage, pathComponents: nil, method: HTTPMethod.GET, parameters: nil)
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil) {
+				completion(dict: JSON.parse(response!))
+			} else {
+				completion(dict: nil)
+			}
+		}
+	}
+	
+	/// Returns the language codes of all available languages that the Zeeguu API supports.
+	///
+	/// - parameter completion: A block that will receive a `JSON` object, which contains the array with the language codes.
+	public func availableLanguages(completion: (array: JSON?) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(array: nil)
+		}
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.AvailableLanguages, pathComponents: nil, method: HTTPMethod.GET, parameters: nil)
+		self.sendAsynchronousRequest(request) { (response, error) -> Void in
+			if (response != nil) {
+				completion(array: JSON.parse(response!))
+			} else {
+				completion(array: nil)
+			}
+		}
+	}
+	
+	/// Returns the words that the user is currently studying.
+	///
+	/// - parameter completion: A block that will receive a `JSON` object, which contains the list of words.
+	public func studyingWords(completion: (array: JSON?) -> Void) {
+		if (!self.checkIfLoggedIn()) {
+			return completion(array: nil)
+		}
+		let request = self.zeeguuAPIRequestWithEndPoint(ZeeguuAPIEndpoint.UserWords, pathComponents: nil, method: HTTPMethod.GET, parameters: nil)
```

Comment:
I like to stay as close to the conventions as possible. The Swift convention is to stick to the Objective C conventions, which results in methods like I created.

Below the Objective-C version the zeeguuAPIRequestWithEndPoint method that I already changed (see new pull request):
```objectivec
- (NSMutableURLRequest *)requestWithEndPoint:(ZeeguuAPIEndpoint)endPoint pathComponents:(NSArray *)pathComponents method:(HTTPMethod)method parameters:(NSDictionary *)parameters;
```
which is the following in Swift (which the new version for now):
```swift
func requestWithEndPoint(endPoint: ZeeguuAPIEndpoint, pathComponents: Array<String>?, method: HTTPMethod, parameters: Dictionary<String, String>?) -> NSURLRequest
```
Although it is possible to use what you'd like (the `endPoint` within the parentheses), I think it is better to stick with the Swift/Objective C conventions.

Also have a look at https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/Swift_Programming_Language/Methods.html
Especially the **Local and External Parameter Names for Methods** and **Modifying External Parameter Name Behavior for Methods** parts.